rfd/0010-api.md

```diff
@@ -37,61 +35,79 @@ func connectClient() (*auth.Client, error) {
     return auth.NewTLSClient(clientConfig)
 ```
 
+It's also impossible to generate library in any other language other than Go due to unclear API surface. Teleport services, auth client, and proto generated files are all dependent on each other in some ways which make a clear distinction difficult. The goal is to make the dependency graph look like `Go stdlib <- proto <- client`, rather than `teleport libraries <- services <- proto <- client`.
+
 ## Details
 
-### Approach
+The api package will be made a Go module, with no dependency on Teleport code outside of itself. This will allow for users to use different Teleport client and server versions together, as long as they are compatibible according to the compatibility matrix. This will also allow the api to have far fewer dependencies, which both lowers the api's import footprint and possible external compatibility issues.
+
+In addition to separating out the client from internal Teleport libraries, the client needs several quality of life improvements. These changes include better client UX via easy credential loading, automatic version checking, better (generated) docs, and more language implementations.
+
+### Make client independent
+
+Keep Teleport's API layer unified, avoid having internal API vs external API layer. Everything needed for a client to use the api client should live within the `api` package.
+
+```
+github.com/gravitational/teleport/api:
+
+go.mod
+go.sum
+api/
+  client/
+    client.go
+    proto/
+      authservice.proto
```

Comment:
The proto file currently includes the service interface which does not belong in the api/client and will probably need to be moved elsewhere

Reply:
Are you referring to the `AuthServiceServer` interface? The `AuthService` defined in the proto file is just the name of the Teleport API as it was,  which connects directly to the auth service. This may be changed in the next step (API credential loading) of the refactor, in which case we may want to rename it to `TeleportAPI` more generally, but I still need to look into this.